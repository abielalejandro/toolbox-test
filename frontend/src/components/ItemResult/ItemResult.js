import { Alert } from "react-bootstrap";

const ItemResult = ({ result }) => {
  return (
    <Alert variant={result.isPalindrome ? "success" : "warning"}>
      <p>Enviado: {result.origin}</p>
      <p>Recibido: {result.text}</p>
      <p>Es palindrome: {result.isPalindrome ? "Si" : "No"}</p>
    </Alert>
  );
};

export default ItemResult;
