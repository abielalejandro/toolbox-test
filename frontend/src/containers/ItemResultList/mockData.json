[
    { "isPalindrome": true, "origin": "ana", "text": "ana" },
    { "isPalindrome": false, "origin": "test", "text": "tset" }
]