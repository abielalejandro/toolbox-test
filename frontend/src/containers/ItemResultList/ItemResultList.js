import PropTypes from "prop-types";
import { Card, Container } from "react-bootstrap";

import ItemResult from "../../components/ItemResult/ItemResult";
const ItemResultList = ({ items }) => {
  return (
    <Container className="d-flex justify-content-center mt-5">
      <Card style={{ width: "60%" }}>
        <Card.Body>
          <Card.Title>Solicitudes</Card.Title>
          {items &&
            items.length > 0 &&
            items.map((item, i) => {
              return <ItemResult key={i} result={item} />;
            })}
        </Card.Body>
      </Card>
    </Container>
  );
};

ItemResultList.defaultProps = {
  items: [],
};

ItemResultList.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      isPalindrome: PropTypes.bool.isRequired,
      text: PropTypes.string.isRequired,
    })
  ),
};

export default ItemResultList;
