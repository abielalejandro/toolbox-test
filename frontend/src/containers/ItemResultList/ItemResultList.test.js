import React from "react";
import { shallow } from "enzyme";
import ItemResultList from "./ItemResultList";
import ItemResult from "../../components/ItemResult/ItemResult";

import mockData from "./mockData.json";

describe("<ItemResultList />", () => {
  it("Expect to not log errors in console", () => {
    // eslint-disable-next-line
    const spy = jest.spyOn(global.console, "error");
    const wrapper = shallow(<ItemResultList items={mockData}></ItemResultList>);
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it("Should render and match the snapshot", () => {
    const wrapper = shallow(<ItemResultList items={mockData}></ItemResultList>);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders items", () => {
    const wrapper = shallow(<ItemResultList items={mockData}></ItemResultList>);
    expect(wrapper.find(ItemResult)).toHaveLength(mockData.length);
  });
});
