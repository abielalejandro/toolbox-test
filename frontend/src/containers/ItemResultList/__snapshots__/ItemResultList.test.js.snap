// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemResult /> Should render and match the snapshot 1`] = `ShallowWrapper {}`;

exports[`<ItemResultList /> Should render and match the snapshot 1`] = `ShallowWrapper {}`;
